plugins/reach/lib/model/EntryVendorTask.php

```diff
@@ -235,9 +240,83 @@ public function getServiceFeature()
 		return $this->getFromCustomData(self::CUSTOM_DATA_SERVICE_FEATURE);
 	}
 
-	/* (non-PHPdoc)
- 	 * @see BaseEntryVendorTask::preSave()
- 	 */
+	protected function calculateNextBusinessDay()
+	{
+		$currentDay = date("N");
+
+		$endDayString = sprintf("Y-m-d %s:00:00", self::BUSINESS_DAY_END_HOUR);
+
+		if( $currentDay > self::BUSINESS_DAY_FRIDAY )
+		{
+			//Weekend
+			$daysToSkip = self::BUSINESS_DAY_NEXT_MONDAY - $currentDay;
+			$expTime = date($endDayString, strtotime("+$daysToSkip days"));
+		}
+		else
+		{
+			$now = date("His");
+
+			$endDayTime = self::BUSINESS_DAY_END_HOUR * self::BUSINESS_DAY_TIME_NORMALIZATION_FACTOR;
+			$startDayTime = self::BUSINESS_DAY_START_HOUR * self::BUSINESS_DAY_TIME_NORMALIZATION_FACTOR;
+
+			//Middle of a business day - expiration time: add 24 hours
+			if( ($startDayTime <= $now) && ($now <= $endDayTime) )
+			{
+				$expTime = date("Y-m-d H:i:s", strtotime('+1 days'));
+			}
+			//Before work hours - expiration time: End Day today
+			elseif ($now < $startDayTime)
+			{
+				$expTime = date($endDayString, strtotime('now'));
+			}
+			//After work hours - expiration time: End Day tomorrow
+			else
+			{
+				$expTime = date($endDayString, strtotime('+1 days'));
+			}
+		}
+
+		return $expTime;
+	}
+
+	protected function calculateBusinessDays($numBusinessDays)
+	{
+		//Get Local TZ
+		$localTimeZone = date_default_timezone_get();
+
+		//Set to EST Time
+		date_default_timezone_set('America/New_York');
+		$currentDateTime = new DateTime("now");
+
+		//Calculate the next business day
+		$expTime = $this->calculateNextBusinessDay();
+
+		//Calculate the second business day
+		if ($numBusinessDays == 2)
```

Comment:
This function should handle also N generic business days and not just 1 / 2, what if we want to add 7 business days?